### PR TITLE
front:fix manchette multiple path management

### LINKED ISSUE #267: error `r2 is undefined` on manchette with many trains
When trying STDCM requests on an imported timetable, the app crash with the error:

```
Unexpected Application Error!
r2 is undefined

it/p2<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:456:15
it/<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:465:70
$/v2</</<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:116:90
$/v2</<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:116:72
$/v2<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:112:7
$/w2<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:119:7
$/<@http://localhost:4000/node_modules/.vite/deps/@osrd-project_ui-manchette.js?v=05f16c15:161:22
commitHookEffectListMount@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:16904:34
commitPassiveMountOnFiber@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:18152:44
commitPassiveMountEffects_complete@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:18125:42
commitPassiveMountEffects_begin@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:18115:49
commitPassiveMountEffects@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:18105:42
flushPassiveEffectsImpl@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:19486:36
flushPassiveEffects@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:19443:22
node_modules/react-dom/cjs/react-dom.development.js/commitRootImpl/<@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:19324:17
workLoop@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:197:50
flushWork@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:176:22
performWorkUntilDeadline@http://localhost:4000/node_modules/.vite/deps/chunk-QYYA6GZJ.js?v=e609e051:384:29

💿 Hey developer 👋

You can provide a way better UX than this when your app throws errors by providing your own ErrorBoundary or errorElement prop on your route.
```

I'm not entirely sure what *exactly* triggers it, it doesn't happen on empty or small timetables. But on imported ones, it happens every time a request is successful.

### LINKED ISSUE #8175: front: space time chart crashes sometimes
### What happened?

I sometimes get `Uncaught TypeError: r2 is undefined`. Not 100% sure how to reproduce consistently, but it happens when clicking around trains in the timetable when they don't all have paths that overlap. Stack trace:

```
Uncaught TypeError: r2 is undefined
p2 PathLayer.tsx:213
it PathLayer.tsx:250
v2 useCanvas.ts:84
v2 useCanvas.ts:84
v2 useCanvas.ts:76
w2 useCanvas.ts:95
$/< useCanvas.ts:194
React 8
workLoop scheduler.development.js:266
flushWork scheduler.development.js:239
performWorkUntilDeadline scheduler.development.js:533
PathLayer.tsx:213:34
```

### What did you expect to happen?

No crash

### How can we reproduce it (as minimally and precisely as possible)?

Not 100% sure how to reproduce consistently, but it happens when clicking around trains in the timetable when they don't all have paths that overlap

### On which environments the bug occurs?

_No response_

### On which browser the bug occurs?

_No response_

### OSRD version (top right corner `Account` button > `Informations`)

.

### LINKED ISSUE #8176: front: space time chart acts odly
### What happened?

There's this short video where I click on "use this path" alternating trains. I don't know exactly what would be the expected behavior, but because the axis don't move and the trains do, the train data *looks* to be changing. But the operation should only change how they're viewed. 

<details>
[Screencast from 24-07-2024 15:45:16.webm](https://github.com/user-attachments/assets/2542d2a0-b8f1-4890-8642-12443fd7f91d)
</details> 

Need investigation

### What did you expect to happen?

The operation should only change how they are viewed.

### How can we reproduce it (as minimally and precisely as possible)?

Reproduce what is on the video

### On which environments the bug occurs?

_No response_

### On which browser the bug occurs?

_No response_

### OSRD version (top right corner `Account` button > `Informations`)

.

### LINKED ISSUE #8189: Cannot open scenario with trains having completely different path
Maybe same problem as in https://github.com/OpenRailAssociation/osrd-ui/issues/267

### What happened?

Import the following timetable (the table has 2 trains, it looks like importing each train separately is ok, but have both trains creates the error)

[schedule bug.json](https://github.com/user-attachments/files/16378105/schedule.bug.json)

then go out of the scenario and go back to the scenario : I have the following error

image

### What did you expect to happen?

I expected to be able to open my scenario

### How can we reproduce it (as minimally and precisely as possible)?

as above

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

7faec4a

### PR DESCRIPTION
**Pull request by Math-R**
close https://github.com/OpenRailAssociation/osrd-ui/issues/267
 close https://github.com/OpenRailAssociation/osrd/issues/8175
 close https://github.com/OpenRailAssociation/osrd/issues/8176
close https://github.com/OpenRailAssociation/osrd/issues/8189

### CODE REVIEW

File: front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts
Comment (clarani) [inline front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts:1]: This file was deleted by error in the previous PR

**Review by clarani — COMMENTED** [review]
LGTM :white_check_mark:
Since I took part in resolving the bug, I will let the other maintainers approve it.

**Review by flomonster — APPROVED** [review]
LGTM: Tested (not reviewed)

**Review by kmer2016 — APPROVED** [review]
LGTM and tested

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested

**Pull request merged**